### PR TITLE
Switch to use the OSS git-sync with githubapp support

### PR DESCRIPTION
**Pull request by nan-yu**
We have been using an internal fork for the githubapp feature. Now this feature is officially included in OSS git-sync v4.3.0. This commit switches to the OSS project for continuous update.

b/376738669

### CODE REVIEW

File: Makefile
Comment (sdowell) [inline Makefile:89]: The base image needs to be bumped before this can be used

File: Makefile
Comment (sdowell) [inline Makefile:320]: I'd consider making this a separate change in case of revert/cherrypick, but no biggie

**Review by sdowell — APPROVED** [review]
/lgtm

File: Makefile
Comment (nan-yu) [reply Makefile:320]: https://github.com/GoogleContainerTools/kpt-config-sync/pull/1471

**Comment by nan-yu** [comment]
Needs a new LGTM after rebasing @sdowell

**Pull request merged**
